src/UsersEpic.js

```diff
@@ -5,18 +5,22 @@ import {map, filter} from 'rxjs/operators';
 import {addUsers} from './UsersAction';
 import {combineEpics} from 'redux-observable';
 import {fetchUsersAC} from './state.js';
+import type {Action} from 'redux';
+import {Observable} from 'rxjs';
+
+let fetchUsersNetworkFails: number = 0;
 
 const fetchUsersCompleteEpic = action$ =>
   action$.pipe(
     ofType(ACTIONS.COMPLETE),
     filter(action => action.payload.name === 'FETCH_USERS'),
     map(({payload}) => {
+      fetchUsersNetworkFails = 0;
```

Comment:
Please research `do` function in rxjs. It's more suitable 
https://stackoverflow.com/questions/40957381/use-case-of-observable-do-operator-rxjs

Reply:
I have used tap in rxjs to do side effects for this epic.